compiler/src/iree/compiler/Dialect/HAL/Transforms/Passes.cpp

```diff
@@ -466,10 +466,6 @@ void buildHALTransformPassPipeline(OpPassManager &passManager,
     passManager.addPass(IREE::HAL::createOutlineMemoizeRegionsPass());
   }
 
-  // If any devices require the legacy synchronous execution behavior then
-  // make all async operations blocking.
-  passManager.addPass(IREE::HAL::createFixupLegacySyncPass());
```

Comment:
Drive by: searching the repo for `legacy.*sync` finds more places to update:

* https://github.com/iree-org/iree/blob/3032df28053e2b401872fce079ba18114c7cef6c/compiler/plugins/target/ROCM/ROCMTarget.cpp#L879-L884
* https://github.com/iree-org/iree/blob/3032df28053e2b401872fce079ba18114c7cef6c/compiler/plugins/target/ROCM/ROCMTarget.cpp#L70
* https://github.com/iree-org/iree/blob/main/compiler/src/iree/compiler/Dialect/HAL/Transforms/test/fixup_legacy_sync.mlir
* https://github.com/iree-org/iree/blob/3032df28053e2b401872fce079ba18114c7cef6c/compiler/src/iree/compiler/Dialect/HAL/Transforms/test/BUILD.bazel#L26

Reply:
Fixed up